crates/router/src/connector/cryptopay/transformers.rs

```diff
@@ -15,21 +16,13 @@ use crate::{
 
 #[derive(Debug, Serialize)]
 pub struct CryptopayRouterData<T> {
-    pub amount: String,
+    pub amount: StringMajorUnit,
     pub router_data: T,
 }
 
-impl<T> TryFrom<(&types::api::CurrencyUnit, enums::Currency, i64, T)> for CryptopayRouterData<T> {
+impl<T> TryFrom<(StringMajorUnit, T)> for CryptopayRouterData<T> {
```

Comment:
we can use a from conversion instead of tryFrom since its 1:1 mapping

Reply:
Done